src/encoder/encodersndfileflac.cpp

```diff
@@ -30,10 +59,38 @@ void EncoderSndfileFlac::setEncoderSettings(const EncoderSettings& settings)
     m_compression = static_cast<double>(settings.getCompression()) / 8.0;
 }
 
+void EncoderSndfileFlac::encodeBuffer(const CSAMPLE* pBuffer, const int iBufferSize) {
+    if (m_pClampBuffer) {
+        SINT numSamplesLeft = iBufferSize;
+        while (numSamplesLeft > kEncBufferSize) {
+            convertFloat32ToIntFormat(m_pClampBuffer.get(),
+                    pBuffer,
+                    kEncBufferSize,
+                    m_sfInfo.format);
+            sf_write_int(m_pSndfile, m_pClampBuffer.get(), kEncBufferSize);
+            pBuffer += kEncBufferSize;
+            numSamplesLeft -= kEncBufferSize;
+        }
+        convertFloat32ToIntFormat(m_pClampBuffer.get(), pBuffer, numSamplesLeft, m_sfInfo.format);
+        sf_write_int(m_pSndfile, m_pClampBuffer.get(), numSamplesLeft);
+    } else {
+        sf_write_float(m_pSndfile, pBuffer, iBufferSize);
+    }
+}
+
 void EncoderSndfileFlac::initStream() {
     EncoderWave::initStream();
 #if defined SFC_SUPPORTS_SET_COMPRESSION_LEVEL // Seems that this only exists since version 1.0.26
     // Tell the compression setting to use.
     sf_command(m_pSndfile, SFC_SET_COMPRESSION_LEVEL, &m_compression, sizeof(double));
 #endif //SFC_SUPPORTS_SET_COMPRESSION_LEVEL
+
+    // Version 1.0.28 suffers broken clamping https://bugs.launchpad.net/mixxx/+bug/1915298
+    // We receive "libsndfile-1.0.28" on Ubuntu Bionic/Focal/Grovy
+    // Older versions are not expected. All newer version have a working internal clamping
+    const char* sf_version = sf_version_string();
+    if (strstr(sf_version, "-1.0.28") != nullptr) {
```

Comment:
How about using the SemanticVersion class?

Reply:
That would rewire to allocate a QString, and also only just works.

Reply:
This is called once on initialization; allocating a QString is no big deal. But I suppose it is unlikely this will be run with an even older version of libsndfile so I don't really care :shrug: